timber-junit/src/main/java/net/lachlanmckee/timberjunit/TimberTestRule.java

```diff
@@ -215,7 +216,9 @@ protected void log(int priority, String tag, String message, Throwable t) {
             }
 
             if (mRules.mOnlyLogWhenTestFails) {
-                mLogMessageBuffer.add(logMessage);
+                synchronized (bufferLock) {
+                    mLogMessageBuffer.add(logMessage);
+                }
 
             } else {
                 System.out.println(logMessage);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)